capstone-app/src/app/modules/maps/map/map.component.spec.ts

```diff
@@ -18,14 +19,64 @@ describe('MapComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(MapComponent);
-    component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
-
   it('should create the app', () => {
     // const fixture = TestBed.createComponent(MapComponent);
     // const app = fixture.componentInstance;
     expect(fixture.componentInstance).toBeTruthy();
   });
+
+  it('test map marker click event', () => {
+    const fixture = TestBed.createComponent(MapComponent);
+    fixture.detectChanges();
+
+    const coordinates = new google.maps.LatLng({lat: 26.011760, lng:  -80.139050});
+
+    fixture.componentInstance.markers.push(coordinates);
+    fixture.detectChanges();
+
+    const markerDebug = fixture.debugElement.query(By.css('.marker'));
+    const markerComponent = markerDebug.componentInstance;
+
+    expect(markerComponent.getPosition()).toEqual(coordinates);
+  });
+
+  it('test multiple map marker click event', () => {
+    const fixture = TestBed.createComponent(MapComponent);
+    fixture.detectChanges();
+    let coordinates: google.maps.LatLng;
+    let lat = 26.011760;
+    let lng = -80.139050
+    for (let i = 0; i < 5; i++) {
+      coordinates = new google.maps.LatLng(lat, lng);
+      lat++;
+      lng++;
+      fixture.componentInstance.markers.push(coordinates);
+      fixture.detectChanges();
```

Comment:
To make this test more efficient, have the fixture detect changes just once after the loop, instead of many times inside the loop.